File descriptor leak in compress mod?

Ain't we leaking fout, if we return here?

https://github.com/eggheads/eggdrop/blob/19b14289acfc504d23e4873fd3834d1ec90de199/src/mod/compress.mod/compress.c#L155

Found while analyzing

https://github.com/wraith/wraith/commit/085bac322731e8a4df8febe22f20689086efc520

Meanwhile, gcc-11-20210207 fanalyzer also reports this leak:
```
/home/michael/opt/gcc-11-20210207/bin/gcc -fanalyzer -fPIC -g -O2 -pipe -Wall -I. -I../../.. -I../../.. -I../../../src/mod  -DHAVE_CONFIG_H -I/usr/include -g3 -DDEBUG -DDEBUG_ASSERT -DDEBUG_MEM -DDEBUG_DNS  -DMAKING_MODS -c .././compress.mod/compress.c && mv -f compress.o ../
In function ‘uncompress_to_file.part.0’:
.././compress.mod/compress.c:155:14: warning: leak of FILE ‘fout’ [CWE-775] [-Wanalyzer-file-leak]
  155 |       return COMPF_ERROR;
      |              ^~~~~~~~~~~
  ‘uncompress_to_file.part.0’: events 1-5
    |
    |  137 |   if (!fin) {
    |      |      ^
    |      |      |
    |      |      (1) following ‘false’ branch (when ‘fin’ is non-NULL)...
    |......
    |  143 |   fout = fopen(f_target, "wb");
    |      |   ~~~~   ~~~~~~~~~~~~~~~~~~~~~
    |      |   |      |
    |      |   |      (3) opened here
    |      |   (2) ...to here
    |  144 |   if (!fout) {
    |      |      ~
    |      |      |
    |      |      (4) assuming ‘fout’ is non-NULL
    |      |      (5) following ‘false’ branch (when ‘fout’ is non-NULL)...
    |
  ‘uncompress_to_file.part.0’: event 6
    |
    |cc1:
    | (6): ...to here
    |
  ‘uncompress_to_file.part.0’: event 7
    |
    |  155 |       return COMPF_ERROR;
    |      |              ^~~~~~~~~~~
    |      |              |
    |      |              (7) ‘fout’ leaks here; was opened at (3)
    |
```

--- a/src/mod/compress.mod/compress.c
+++ b/src/mod/compress.mod/compress.c
@@ -152,6 +152,7 @@ static int uncompress_to_file(char *f_src, char *f_target)
     if (len < 0) {
       putlog(LOG_MISC, "*", "Failed to uncompress file `%s': gzread failed.",
              f_src);
+      fclose(fout);
       return COMPF_ERROR;
     }
     if (!len)
@@ -159,6 +160,7 @@ static int uncompress_to_file(char *f_src, char *f_target)
     if ((int) fwrite(buf, 1, (unsigned int) len, fout) != len) {
       putlog(LOG_MISC, "*", "Failed to uncompress file `%s': fwrite "
              "failed: %s.", f_src, strerror(errno));
+      fclose(fout);
       return COMPF_ERROR;
     }
   }

